bolt-simple-taproot.md

```diff
@@ -0,0 +1,943 @@
+    2: data:
+        * [`66*byte`:`nonces`]
+
+##### Requirements
+
+The sending node:
+
+  - MUST specify the `local_musig2_pubnonce` and `remote_musig2_pubnonce` TLV
+    fields.
+
+The sending node SHOULD:
+
+  - use the `NonceGen` algorithm defined in `bip-musig2` to generate
+    `lcaol_musig2_pubnonce` and `remote_musig2_pubnonce` to  ensure it
+    generates nonces in a safe manner.
+
+The receiving node MUST fail the channel if:
+
+  - the message doesn't include a `local_musig2_pubnonce` and
+    `remote_musig2_pubnonce` value.
+
+  - a specified public nonce cannot be parsed as two compressed secp256k1
+    points
+
+##### Rationale
+
+For the initial Taproot channel type (`option_simple_taproot`), musig2 nonces
+must be exchanged in the first round trip (open->, <-accept) to ensure that the
+initiator is able to generate a valid musig2 partial signature at the next
+step.
+
+The `remote_musig2_pubnonce` will be used to sign the _remote_ party's
+commitment transaction, while the `local_musig2_pubnonce` will be used to sign
+the local party's commitment transaction. 
+
+We require two nonces, as there're actually two messages being signed: the
+local commitment and the remote commitment. If only one nonce was used, when a
+party went to sign and broadcast their local commitment, they would re-use a
+nonce, as the nonce was already used to sign the remote party's commitment.
+
+#### `accept_channel` Extensions
+
+1. `tlv_stream`: `accept_channel_tlvs`
+2. types:
+    1. type: 2 (`local_musig2_pubnonce`)
+    2: data:
+        * [`66*byte`:`nonces`]
+    1. type: 4 (`remote_musig2_pubnonce`)
+    2: data:
+        * [`66*byte`:`nonces`]
+
+##### Requirements
+
+The sender:
+
+ - MUST set `local_musig2_pubnonce` and `remote_musig2_pubnonce` to the
+   `musig2` public nonce as specified by the `NonceGen` algorithm of
+   `bip-musig2`
+
+ - Aggregate the received local+remote public nonces (into two combined nonces)
+   using the `NonceAgg` algorithm of `bip-musig2`.
+
+The receiver:
+
+ - MUST reject the channel if `local_musig2_pubnonce` or
+   `remote_musig2_pubnonce` are not present.
+
+#### `funding_created` Extensions
+
+The sender:
+
+  - MUST derive the aggregated `musig2` public key using the exchanged `funding_pubkey`s 
+    - Before aggregating the two funding public keys, the keys MUST be sorted
+      using the `KeySort` routine of `bip-musig2`.
+
+  - MUST use the exchanged `local_musig2_pubnonce` and `remote_musig2_pubnonce`
+    received from the remote party with the `NonceAgg` algorithm to derive the
+    combined nonce
+
+  - the `signature` field MUST be a `musig2` partial signature using the
+    aggregated key defined above, and the two public nonces exchanged in prior
+    messages.
+
+  - MUST generate the attached `signature` using the combined
+    `remote_musig2_pubnonce` it generated in the prior step.
+
+
+The recipient:
+  - MUST validate the signature as a _partial_ signature based on the
```

Comment:
partial sigs are 32 bytes, not 64 like full sigs. Are your requiring the public nonces to be re-sent or should we break this into a new message type that's more precisely defined?

Reply:
See this comment: https://github.com/lightning/bolts/pull/995#issuecomment-1170703114

Main motivation to changing it is sending less bytes I guess? Tho it pales in comparison to how large the onoin messages are.